test/import-style.js

```diff
@@ -0,0 +1,550 @@
+		},
+		{
+			code: 'const {x: y} = require(\'namespace\')',
+			errors: [namespaceError]
+		},
+		{
+			code: 'import {x} from \'namespace\'',
+			errors: [namespaceError]
+		},
+		{
+			code: 'import {x as y} from \'namespace\'',
+			errors: [namespaceError]
+		},
+		{
+			code: outdent`
+				async () => {
+					const {x} = await import('namespace');
+				}
+			`,
+			errors: [namespaceError]
+		},
+		{
+			code: outdent`
+				async () => {
+					const {x: y} = await import('namespace');
+				}
+			`,
+			errors: [namespaceError]
+		},
+		{
+			code: 'export {x} from \'namespace\'',
+			errors: [namespaceError]
+		},
+		{
+			code: 'export {x as y} from \'namespace\'',
+			errors: [namespaceError]
+		},
+		{
+			code: 'export {default} from \'namespace\'',
+			errors: [namespaceError]
+		},
+
+		{
+			code: 'require(\'named\')',
+			errors: [namedError]
+		},
+		{
+			code: 'import \'named\'',
+			errors: [namedError]
+		},
+		{
+			code: 'import(\'named\')',
+			errors: [namedError]
+		},
+		{
+			code: 'const x = require(\'named\')',
+			errors: [namedError]
+		},
+		{
+			code: 'const {default: x} = require(\'named\')',
+			errors: [namedError]
+		},
+		{
+			code: 'import x from \'named\'',
+			errors: [namedError]
+		},
+		{
+			code: outdent`
+				async () => {
+					const {default: x} = await import('named');
+				}
+			`,
+			errors: [namedError]
+		},
+		{
+			code: 'import * as x from \'named\'',
+			errors: [namedError]
+		},
+		{
+			code: outdent`
+				async () => {
+					const x = await import('named');
+				}
+			`,
+			errors: [namedError]
+		},
+		{
+			code: 'export * from \'named\'',
+			errors: [namedError]
+		},
+		{
+			code: 'export {default} from \'named\'',
+			errors: [namedError]
+		},
+
+		{
+			code: 'import util, {inspect} from \'named\'',
+			errors: [namedError]
+		},
+		{
+			code: 'import util, {inspect} from \'default\'',
+			errors: [defaultError]
+		},
+
+		{
+			code: 'import util from \'util\'',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: 'import * as util from \'util\'',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: 'const util = require(\'util\')',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: 'require(\'util\')',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: 'import {red} from \'chalk\'',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: 'import {red as green} from \'chalk\'',
+			options: [],
+			errors: [{}]
+		},
+		{
+			code: outdent`
+				async () => {
+					const {red} = await import('chalk');
+				}
+			`,
+			options: [],
+			errors: [{}]
+		},
+
+		{
+			code: 'require(\'no-unassigned\')',
+			options: [{
+				styles: {
+					'no-unassigned': {
+						named: true,
+						namespace: true,
+						default: true
+					}
+				}
+			}],
+			errors: [{
+				messageId: 'importStyle',
+				data: {
+					allowedStyles: 'named, namespace or default',
+					moduleName: 'no-unassigned'
+				}
+			}]
+		}
+	].map(test => addDefaultOptions(test))
+});
```

Comment:
Let's test 

```js
const {} = require('foo')
```

and

```js
import {} from 'foo'
```

too, they are valid code, just make sure we don't crash on them.